src/input/composite_device/mod.rs

```diff
@@ -1479,6 +1479,23 @@ impl CompositeDevice {
         let path = device.devnode();
         let id = device.get_id();
 
+        let mut freed_effect_ids = Vec::new();
+
```

Comment:
Add Comment
`// Loop through all registered effects and remove this device as a source`